dm/dm/master/scheduler/scheduler.go

```diff
@@ -694,11 +753,22 @@ func (s *Scheduler) TransferSource(source, worker string) error {
 		s.logger.DPanic("we have checked w.stage is free, so there should not be an error", zap.Error(err2))
 	}
 
-	// 6. try bound the old worker
+	// 6. now this old worker is free, try bound source to it
 	_, err = s.tryBoundForWorker(oldWorker)
 	if err != nil {
 		s.logger.Warn("in transfer source, error when try bound the old worker", zap.Error(err))
 	}
+	// we need resume tasks that we just paused, we use another goroutine to do this because if error happens
+	// just logging this message and let user handle it manually
+	go func() {
+		for _, task := range runningTasks {
+			if err := s.UpdateExpectSubTaskStage(pb.Stage_Running, task, source); err != nil {
+				s.logger.Warn(
+					"auto resume task failed", zap.String("task", task),
+					zap.String("source", source), zap.String("worker", worker), zap.Error(err))
+			}
+		}
+	}()
```

Comment:
Will this background function affect other operations? Since we didn't hold write lock any more in this func.

Reply:
thanks! we need add lock for this operation 
added in https://github.com/pingcap/tiflow/pull/3889/commits/0dbd8c48417c5caa6c8f24707f4caa9eb5ce3b1e and https://github.com/pingcap/tiflow/pull/3889/commits/7549c4d2fe4c51504b9c0c6b66877beba517bf23

Reply:
we not use`s.mu` to lock shcheduler but relay on `subtaskLatch` see more in this [comment](https://github.com/pingcap/tiflow/pull/3889#discussion_r773711744)